Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -1325,42 +1325,80 @@ extension Tensor.IndexPath {
 //===------------------------------------------------------------------------------------------===//
 
 extension Tensor {
-  /// Returns `true` if the given axis is in the range `[-rank, rank)`.
+  /// Returns `true` iff `k` denotes an axis of `self`.
   @usableFromInline
-  internal func isAxisInRange<T: BinaryInteger>(_ axis: T) -> Bool {
-    let axis = Int(axis)
+  internal func isValid<T: BinaryInteger>(axis k: T) -> Bool {
+    let axis = Int(k)
     return axis >= -rank && axis < rank
   }
 
-  /// Returns `true` if the given scalar tensor is in the range `[-rank, rank)`.
+  /// Returns `true` iff each element of `dims` denotes an axis of `self`.
   @usableFromInline
-  internal func isAxisInRange(
-    _ axis: Tensor<Int32>,
+  internal func areValid<T: BinaryInteger>(axes dims: [T]) -> Bool {
+    return dims.allSatisfy { isValid(axis: $0) }
+  }
+
+  /// Returns `true` iff each element of `dims` denotes an axis of `self`.
+  ///
+  /// - Precondition: `dims` is rank 0 or rank 1.
```

Comment:
```suggestion
  /// - Precondition: `dims` has rank 0 or rank 1.
```

or

```suggestion
  /// - Precondition: `(0...1).contains(dims.rank`
```

The first one is probably more natural.

Reply:
Definitely strongly prefer the first one. Thanks!